lib/source/common/iot_taskpool.c

```diff
@@ -44,13 +44,13 @@
  * @brief Enter a critical section by locking a mutex.
  *
  */
-#define TASKPOOL_ENTER_CRITICAL()        IotMutex_Lock( &( ( ( IotTaskPool_t * ) pTaskPool )->lock ) )
+#define TASKPOOL_ENTER_CRITICAL()        IotMutex_Lock( &( ( ( _taskPool_t *) pTaskPool )->lock ) )
```

Comment:
Is this cast needed?  Taskpool is already typedef to _taskPool_t *

Reply:
Fixed